docs/readmes/contributing/contribute_conventions.md

```diff
@@ -0,0 +1,186 @@
+---
+id: contribute_conventions
+title: Code Conventions
+hide_title: true
+---
+
+# Code Conventions
+
+This document describes code conventions for the Magma project. The goal of this style guide is to
+
+- Steer contributors away from experienced landmines
+- Align on coding styles in support of a uniform Magma codebase, with the aim to improve developer productivity and codebase approachability
+
+## General
+
+Throughout the Magma codebase, the principal convention is the [boy scout rule](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html): leave it better than how you found it. Consider the following conventions when adding a PR
+
+- Add tests for your changes
+    - Tests should cover, at minimum, the mainline of the new feature. For reference, that usually ends up around 50-70% coverage.
+    - Tests should be in the same directory as the files they're testing
+        - Exception: Python and C/C++ convention is to use directly-adjacent `tests` directory
```

Comment:
Considering AGW is really running on C/C++ and Python, I would not categorize this as an exception.